File: SmartInventoryAPI/Models/Entities/Vendor.cs
namespace SmartInventoryAPI.Models.Entities;

public class Vendor
{
    public long ID { get; set; }
    public string? Company_Name { get; set; }
    public string? Vendor_Code { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime Creation_Date { get; set; } = DateTime.UtcNow;
    public string? Company_Address { get; set; }
    public string? Email { get; set; }
    public string? Mobile { get; set; }
}

File: SmartInventoryFrontend/src/types/vendor.ts
import { Entity } from './common'

export interface Vendor extends Entity {
  company_Name?: string
  vendor_Name?: string
  vendor_Code?: string
  company_Address?: string
  address?: string
  email?: string
  phone_No?: string
  mobile?: string
}

export interface CreateVendorRequest {
  company_Name: string
  vendor_Code: string
  company_Address: string
  email: string
  mobile: string
}

export interface UpdateVendorRequest {
  company_Name?: string
  vendor_Code?: string
  company_Address?: string
  email?: string
  mobile?: string
}

export interface VendorState {
  vendors: Vendor[]
  currentVendor: Vendor | null
  loading: boolean
  error: string | null
  total: number
  skip: number
  take: number
}
